src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs

```diff
@@ -49,7 +49,16 @@ private StreamFormatter(MessageDescription messageDescription, MessagePartDescri
         internal void Serialize(XmlDictionaryWriter writer, object[] parameters, object returnValue)
         {
             Stream streamValue = GetStreamAndWriteStartWrapperIfNecessary(writer, parameters, returnValue);
-            writer.WriteValue(new OperationStreamProvider(streamValue));
+            var streamProvider = new OperationStreamProvider(streamValue);
+            StreamFormatterHelper.WriteValue(writer, streamProvider);
+            WriteEndWrapperIfNecessary(writer);
+        }
+
+        internal async Task SerializeAsync(XmlDictionaryWriter writer, object[] parameters, object returnValue)
+        {
+            Stream streamValue = GetStreamAndWriteStartWrapperIfNecessary(writer, parameters, returnValue);
```

Comment:
Should we have an async variation of GetStreamAndWriteStartWrapperIfNecessary? Not for this change, but if we should (it looks like we're writing to the stream in that method), then make sure this is recorded an a relevant bug somewhere.